### PR TITLE
Track Arbitrum finality

### PR DESCRIPTION
**Pull request by michalsidzej**
Resolves L2B-4606

### CODE REVIEW

File: packages/backend/src/modules/finality/analyzers/arbitrum/blobsToData.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  }
+
+  const rlpData = new Uint8Array(data.reduce((acc, val) => acc + val.length, 0))
+  let offset = 0
+  for (const val of data) {
+    rlpData.set(val, offset)
+    offset += val.length
+  }
+
+  const [decoded] = rlpDecodePartial(rlpData)
```
Comment (mateuszradomski) [inline packages/backend/src/modules/finality/analyzers/arbitrum/blobsToData.ts:52]: Why partial?

File: packages/backend/src/modules/finality/analyzers/arbitrum/getSegments.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  const decompressed = decompressPayload(payload)
+  const rlpEncoded = concatWithLengthForRlp(decompressed)
+  const segments = rlpDecode(rlpEncoded)
+  assert(Array.isArray(segments), 'Expected segments to be an array')
+
+  return segments
+}
+
+export function decompressPayload(payload: Uint8Array): Uint8Array {
+  // todo: we should check the first byte. It's probably the compression type.
```
Comment (mateuszradomski) [inline packages/backend/src/modules/finality/analyzers/arbitrum/getSegments.ts:21]: It is. Should always be `0x00` for brotli compression

File: packages/backend/src/modules/finality/analyzers/arbitrum/getSegments.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  const data = payload.slice(1)
+  const decompressedData = zlib.brotliDecompressSync(data)
+  return Uint8Array.from(decompressedData)
+}
+
+export function concatWithLengthForRlp(decompressedBytes: Uint8Array) {
```
Comment (mateuszradomski) [inline packages/backend/src/modules/finality/analyzers/arbitrum/getSegments.ts:27]: image

File: packages/backend/src/modules/finality/analyzers/arbitrum/getSegments.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import { assert } from '@l2beat/shared-pure'
+import zlib from 'zlib'
+
+import { rlpDecode } from '../../utils/rlpDecode'
+import { byteArrFromHexStr } from '../opStack/utils'
+import { blobsToData } from './blobsToData'
+import { numberToByteArr } from './utils'
+
+export function getSegments(relevantBlobs: { blob: string }[]) {
```
Comment (mateuszradomski) [inline packages/backend/src/modules/finality/analyzers/arbitrum/getSegments.ts:9]: We should type the return type so that we don't pass `unknown` around.

**Review by mateuszradomski — CHANGES_REQUESTED** [review]

File: packages/backend/src/modules/finality/analyzers/arbitrum/getSegments.ts (hunk 3 above)
Comment (michalsidzej) [reply packages/backend/src/modules/finality/analyzers/arbitrum/getSegments.ts:27]: I literally have no idea, but it works.

File: packages/backend/src/modules/finality/analyzers/arbitrum/getSegments.ts (hunk 2 above)
Comment (michalsidzej) [reply packages/backend/src/modules/finality/analyzers/arbitrum/getSegments.ts:21]: That's why I like your reviews

File: packages/backend/src/modules/finality/FinalityIndexer.test.ts [hunk 5]
```diff
@@ -81,6 +81,27 @@ describe(FinalityIndexer.name, () => {
...
+        finalityRepository,
+      })
+      finalityIndexer.isConfigurationSynced = mockFn().resolvesToOnce(false)
+
+      const start = UnixTime.now().add(-3, 'days')
+
+      const from = start.toNumber()
+      const to = start.add(1, 'days').toNumber()
+
+      // TODO: refactor tests after uif update
```
[the remark this answers is not in the document]
Comment (michalsidzej) [reply packages/backend/src/modules/finality/FinalityIndexer.test.ts:101]: [56638c9](https://github.com/l2beat/l2beat/pull/3288/commits/56638c9df79ffa9adabafba9fc6ee50d8a45b561)

File: packages/backend/src/modules/finality/analyzers/arbitrum/blobsToData.ts (hunk 1 above)
Comment (michalsidzej) [reply packages/backend/src/modules/finality/analyzers/arbitrum/blobsToData.ts]: [chore: explain code to the reviewers](https://github.com/l2beat/l2beat/pull/3288/commits/b68a3822f6b699bbcb8be92ea966c620f0e0d88a)

File: packages/backend/src/modules/finality/analyzers/arbitrum/getSegments.ts (hunk 3 above)
Comment (michalsidzej) [reply packages/backend/src/modules/finality/analyzers/arbitrum/getSegments.ts:37]: [chore: more comments](https://github.com/l2beat/l2beat/pull/3288/commits/2bde723ec3861ea94166cc861e64895aac846b0c)

File: packages/backend/src/modules/finality/analyzers/arbitrum/getSegments.ts (hunk 2 above)
Comment (michalsidzej) [reply packages/backend/src/modules/finality/analyzers/arbitrum/getSegments.ts]: [b68a382](https://github.com/l2beat/l2beat/pull/3288/commits/b68a3822f6b699bbcb8be92ea966c620f0e0d88a)

**Review by mateuszradomski — APPROVED** [review]

**Review by maciekzygmunt — APPROVED** [review]

**Pull request merged**